### ISSUE TITLE
Control statement rule doesn't handle precedence parens

### ISSUE THREAD

**Issue by keith** [realm/SwiftLint#89]
Currently, [this validator](https://github.com/realm/SwiftLint/blob/047922610b67c125f57a5fc87cffb115f932d66a/Source/SwiftLintFramework/Rules/ControlStatementRule.swift#L16-L30) doesn't handle parens used for precendence. For example:

```swift
if (a && b) || (c && d) {
```

Currently, using [mayday](https://github.com/marklarr/mayday) you can fix this with recursive regex :trollface::

```ruby
warning_regex "No parens around if statements",
  /if\s+(?<match>\((?:\g<match>|[^()]++)*\))\s*{/, language: :swift
```

But it doesn't seem like `NSRegularExpression` supports this.

**Comment by jpsim** [comment]
Well, that's unfortunate! Anyone care to build a recursive regular expression library in Swift? :trollface:

**Comment by neilpa** [comment]
By definition regular expressions can't be recursive :trollface:

**Comment by jpsim** [comment]
Fixed in #195.
